e2e/awspcaissuer_test.go

```diff
@@ -0,0 +1,237 @@
+	"github.com/cert-manager/aws-privateca-issuer/pkg/api/v1beta1"
+	clientV1beta1 "github.com/cert-manager/aws-privateca-issuer/pkg/clientset/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var testContext *TestContext
+
+// These are variables shared by all of the tests
+type TestContext struct {
+	iclient   *clientV1beta1.Client
+	cmClient  *cmclientv1.CertmanagerV1Client
+	clientset *kubernetes.Clientset
+	xaCfg     aws.Config
+	caArns    map[string]string
+
+	region, accessKey, secretKey, endEntityResourceShareArn, subordinateCaResourceShareArn, userName, policyArn string
+}
+
+// These are variables specific to each test
+type IssuerContext struct {
+	certName   string
+	issuerName string
+	issuerType string
+	namespace  string
+	secretRef  v1beta1.AWSCredentialsSecretReference
+}
+
+var opts = godog.Options{
+	Concurrency: 8,
+	Format:      "pretty",
+	Paths:       []string{"features"},
+}
+
+func TestMain(m *testing.M) {
+	o := opts
+
+	_, xaRoleExists := os.LookupEnv("PLUGIN_CROSS_ACCOUNT_ROLE")
+	if !xaRoleExists {
+		log.Printf("Skipping CrossAccount tests")
+		o.Tags = "~@CrossAccount"
+	}
+	status := godog.TestSuite{
+		Name:                 "AWSPrivateCAIssuer",
+		Options:              &o,
+		ScenarioInitializer:  InitializeScenario,
+		TestSuiteInitializer: InitializeTestSuite,
+	}.Run()
+
+	os.Exit(status)
+}
+
+func InitializeTestSuite(suiteCtx *godog.TestSuiteContext) {
+	// This is a BeforeAll hook that initializes the variables in the TestContext struct
+	suiteCtx.BeforeSuite(func() {
+		ctx := context.TODO()
+		testContext = &TestContext{}
+		testContext.caArns = make(map[string]string)
+
+		//setup k8 client
+		//kubeconfig files will be gathered from the home directory
+		// tmp/pca_kubeconfig is auto populated if creating cluster from makefile
+		kubeconfig := "/tmp/pca_kubeconfig"
+
+		clientConfig, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			panic("Ensure that that the kubeconfig for the cluster that is being tested is placed in /tmp/pca_kubeconfig")
+		}
+
+		testContext.clientset, err = kubernetes.NewForConfig(clientConfig)
+		if err != nil {
+			panic(err.Error())
+		}
+
+		// We set the region to the AWS_REGION environment variable or us-east-1 by default
+		testContext.region = "us-east-1"
+		region, regionExists := os.LookupEnv("AWS_REGION")
+		if regionExists {
+			testContext.region = region
+		}
+
+		cfg, cfgErr := config.LoadDefaultConfig(ctx, config.WithRegion(testContext.region))
+		if cfgErr != nil {
+			panic(cfgErr.Error())
+		}
+
+		testContext.iclient, err = clientV1beta1.NewForConfig(clientConfig)
+
+		if err != nil {
+			panic(err.Error())
+		}
+
+		testContext.cmClient, err = cmclientv1.NewForConfig(clientConfig)
+
+		if err != nil {
+			panic(err.Error())
+		}
+
+		// Create CAs to be used in testing
+		testContext.caArns["RSA"] = createCertificateAuthority(ctx, cfg, true)
```

Comment:
Can we use ACK now! Please.

Reply:
Example here: https://github.com/aws-samples/sample-encryption-in-transit-for-kubernetes/blob/main/deploy-core-pki/deploy-core.sh#L43-L70